nav2_collision_monitor/src/collision_monitor_node.cpp

```diff
@@ -68,11 +72,16 @@ CollisionMonitor::on_configure(const rclcpp_lifecycle::State & /*state*/)
     std::bind(&CollisionMonitor::cmdVelInCallback, this, std::placeholders::_1));
   cmd_vel_out_pub_ = this->create_publisher<geometry_msgs::msg::Twist>(
     cmd_vel_out_topic, 1);
+
   if (!state_topic.empty()) {
     state_pub_ = this->create_publisher<nav2_msgs::msg::CollisionMonitorState>(
       state_topic, 1);
   }
 
+  if (visualize_collision_points) {
```

Comment:
Why return? Just store as a class member and grab it here. Lets not complicate the APIs (and would make this not backportable since you change ABI!!)

Reply:
I would have done it like you proposed as well but that's how it's done for all other parameters of the collision monitor like `cmd_vel_in_topic`, `cmd_vel_out_topic`, `state_topic`. So what do you prefer, breaking ABI or introducing some inconsistency?

Reply:
See new comments below, this may be moot